receiver/httpreceiver/config.go

```diff
@@ -0,0 +1,65 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package httpreceiver is a default HTTP receiver for log ingestion
+package httpreceiver
+
+import (
+	"errors"
+	"net"
+	"path"
+
+	"go.opentelemetry.io/collector/config/confighttp"
+)
+
+// Config defines the configuration for an HTTP receiver
+type Config struct {
+	Path           string                         `mapstructure:"path"`
+	ServerSettings *confighttp.HTTPServerSettings `mapstructure:"server"`
```

Comment:
Small recommendation here:
```suggestion
	ServerSettings *confighttp.HTTPServerSettings `mapstructure:",squash"`
```

This way we don't need the extra `server` key when specifying endpoint/tls. I think this is the usual pattern with this struct (e.g. https://github.com/open-telemetry/opentelemetry-collector/blob/2a9d870a57038cb228ef20e7620f18a93c92b9d3/receiver/otlpreceiver/config.go#L25)

Reply:
oh yea perfect, didn't realize i could do that